components/telegram_bot_api/telegram_bot_api.app.js

```diff
@@ -89,5 +271,386 @@ module.exports = {
+    },
+    /**
+     * Delete a message
+     *
+     * @param  {Number|String} chatId - Unique identifier of the target chat
+     * @param  {Number} messageId - Unique identifier of the target message
+     * @returns `True` on success
+     */
+    async deleteMessage(chatId, messageId) {
+      return await this.sdk().deleteMessage(chatId, messageId);
+    },
+    /**
+     * Use this method to add a message to the list of pinned messages in a
+     * chat. If the chat is not a private chat, the bot must be an administrator
+     * in the chat for this to work and must have the 'can_pin_messages' admin
+     * right in a supergroup or 'can_edit_messages' admin right in a channel.
+     *
+     * @param {Number|String} chatId - Unique identifier for the message
+     * recipient
+     * @param {Number} messageId - Identifier of a message to pin
+     * @param {Object} [opts] - Additional Telegram query options to be fed to
+     * the Telegram Bot API call, as defined in [the API
+     * docs](https://core.telegram.org/bots/api#pinchatmessage)
+     * @return `True` on success
+     */
+    async pinChatMessage(chatId, messageId, opts) {
+      return await this.sdk().pinChatMessage(chatId, messageId, opts);
+    },
+    /**
+     * Use this method to remove a message from the list of pinned messages in a
+     * chat. If the chat is not a private chat, the bot must be an administrator
+     * in the chat for this to work and must have the 'can_pin_messages' admin
+     * right in a supergroup or 'can_edit_messages' admin right in a channel.
+     *
+     * @param {Number|String} chatId - Unique identifier for the message
+     * recipient
+     * @param {Object} [opts] - Additional Telegram query options to be fed to
+     * the Telegram Bot API call, as defined in [the API
+     * docs](https://core.telegram.org/bots/api#unpinchatmessage)
+     * @return `True` on success
+     */
+    async unpinChatMessage(chatId, messageId) {
+      return await this.sdk().unpinChatMessage(chatId, {
+        message_id: messageId,
+      });
+    },
+    /**
+     * @typedef {function} SendMediaFn
+     * @param {string} chatId - Unique identifier for the target chat or
+     * username of the target channel (in the format @channelusername)
+     * @param {String|stream.Stream|Buffer} media - A file path or a Stream. Can
+     * also be a `file_id` previously uploaded
+     * @param {Object} [opts] - An object containing additional configuration
+     * options for this method
+     * @param {Number|String} [opts.filename] - The name of the file to send
+     * @param {Number|String} [opts.contentType] - The MIME type of the file to
+     * send
+     * @param {...*} [opts.extraOpts] - Additional Telegram query options to be
+     * fed to the Telegram Bot API call, as defined in [the API
+     * docs](https://core.telegram.org/bots/api)
+     * @returns {Promise<TelegramBot.Message>} The sent message
+    */
```

Comment:
Not sure what is the reason you put two versions of the jsdoc?

Reply:
I put two version of the jsdoc because [this version](https://github.com/PipedreamHQ/pipedream/blob/3f114c56dad3346420f7888ae5357e9b1f45ab85/components/telegram_bot_api/telegram_bot_api.app.js#L408) has an additional `sendFn` param. Ideally docs for the params wouldn't be repeated, but I'm not sure how to avoid duplicating docs for functions with mostly the same params. Maybe these methods could be refactored so they have the same arguments.

Reply:
I still don't get it maybe I'm a little slow here. But that means I can call `sendMedia` without `sendFn` function? Even though you are only calling it passing `sendFn` right?

Reply:
I think a practical way to solve this is by doing what I suggested below with my opinon. But again that's just an opinion.

Reply:
I removed one of the versions of the jsdoc.